hack/tools/generate_versions.go

```diff
@@ -0,0 +1,212 @@
+package main
```

Comment:
It'd be great if we gradually migrate the other `.sh` scripts into Go for more readability and sanity :)
I'll create a ticket

Tell me if comments are needed for thei one.

Reply:
I'd rather keep hack/tools for tracking/installing external Go binaries that are needed by the build process. Can we move the program to a dedicated subdirectory?

Reply:
You're right, I thought I put them next to the sh ones...
I put them in a go/ (not too creative :)) dir now.